### DISCUSSION TITLE
Uploading multiple files to supabase at once.

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [supabase/supabase#6101] [Questions] [answered]
Hello,
Is there a way I can upload multiple files to supabase storage at once or do I have to loop through the files and upload each one individually?

**Comment by GaryAustin1** [comment, marked as answer]
You will need to loop.

**Reply by Electric-Dragon** [reply]
Yep okay thanks

**Comment by JeremyMees** [comment]
Is there still no better answer to this?
I have it working with a loop but i would be nice if its just on call

**Comment by GaryAustin1** [comment]
@JeremyMees This was opened as a feature request and closed in the linked issue a few weeks ago.  s3 does not support batch uploads.  So basically someone has to do the loop, and for now that is you...

https://github.com/supabase/storage-api/issues/166

**Reply by oxneat** [reply]
@GaryAustin1 😂 This is Helarious

**Reply by firstpersoncode** [reply]
that is you...... 😂

**Reply by dalindev** [reply]
geez, I almost choked drinking my coffee while reading this... my ChatGPT-4 will do the loop then, thanks!

**Reply by IbraheemHaseeb7** [reply]
time to launch an npm package 😂

**Comment by Shaukat456** [comment]
What is the the maximum number of files I can upload in a batch.


![image](https://github.com/supabase/supabase/assets/65353819/cb609f3f-ced6-400c-9083-e6f67cbc43d6)


How to determine the exact number where it fails?
